tests/__tests__/tsconfig-default.spec.ts

```diff
@@ -13,78 +15,55 @@ describe('get default ts config', () => {
 
-  it('should not read my-tsconfig.json', () => {
-    const result = getTSConfig(null);
-
-    expect(result).not.toEqual({
-      target: ts.ScriptTarget.ES2015,
-      module: ts.ModuleKind.CommonJS,
-      moduleResolution: ts.ModuleResolutionKind.NodeJs,
-      noEmitOnError: true,
-      jsx: ts.JsxEmit.React,
-    });
+    expect(() => getTSConfig(jestConfig.jestconfigTest(null))).toThrowError(
+      /unable to find ts configuration file/i,
+    );
   });
 
-  it('should not read inline tsconfig options', () => {
-    const result = getTSConfig(null);
+  it('should correctly read tsconfig.json', () => {
+    const result = getTSConfig(jestConfig.tsconfigTest(null));
 
-    expect(result).not.toEqual({
-      module: ts.ModuleKind.CommonJS,
-      jsx: ts.JsxEmit.React,
-    });
+    expect(result).toMatchSnapshot();
   });
 
   describe('new behavior (tsConfigFile & tsConfig)', () => {
     it('should be same results for null/undefined/etc.', () => {
-      const result = getTSConfig(null);
-      const resultUndefinedParam = getTSConfig(undefined);
-      const resultNullParam = getTSConfig(null);
-      const resultEmptyParam = getTSConfig({});
-      const resultUndefinedContentFile = getTSConfig({
-        'ts-jest': { tsConfigFile: undefined },
-      });
-      const resultNullContentFile = getTSConfig({
-        'ts-jest': { tsConfigFile: null },
-      });
-
-      expect(result).toEqual(resultUndefinedParam);
-      expect(result).toEqual(resultNullParam);
+      const result = getTSConfig(jestConfig.tsconfigTest(null));
+      const resultEmptyParam = getTSConfig(jestConfig.tsconfigTest({}));
+      const resultUndefinedContentFile = getTSConfig(
+        jestConfig.tsconfigTest({ tsConfigFile: undefined }),
+      );
+      const resultNullContentFile = getTSConfig(
+        jestConfig.tsconfigTest({ tsConfigFile: null }),
+      );
+
       expect(result).toEqual(resultEmptyParam);
       expect(result).toEqual(resultUndefinedContentFile);
       expect(result).toEqual(resultNullContentFile);
     });
 
     it('should be different results for different rootDir with same jest config.', () => {
-      const rootConfig = getTSConfig({});
-      const subConfig = getTSConfig({}, 'tsconfig-module');
+      const rootConfig = getTSConfig(jestConfig.tsconfigTest());
+      const subConfig = getTSConfig(
+        jestConfig('tsconfig-test/tsconfig-module'),
+      );
       expect(rootConfig).not.toEqual(subConfig);
     });
 
     it('should not change the module if it is loaded from a non-default config file', () => {
-      const config = getTSConfig({
-        'ts-jest': {
+      const config = getTSConfig(
+        jestConfig.tsconfigTest({
           tsConfigFile: 'tsconfig-module/custom-config.json',
-        },
-      });
+        }),
+      );
 
+      // snapshot would be enough here, but that adds a security in case we do not see it in a PR
       expect(config.module).toBe(ts.ModuleKind.ES2015);
```

Comment:
CI won't pass if there's a snapshot error.

Reply:
Yup, but this is because @kulshekhar was concerned about us not seeing a snapshot change in a PR, like if we have done a `jest ... -u` and did not check the snapshot diff. Here it's a little test ahead that would fail anyway, even with the `-u`. It's a security in case we hadn't seen the diff of the related snapshot.

Reply:
Ah okay. I don't think I share his fears, but I don't see any harm in leaving it in.